examples/image_captioning/image_captioning.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+"""
+Image Captioning with Ludwig
+
+In this notebook, we will use ludwig to train an image captioning system. 
+We be using the Flickr 8k dataset which contains eight thousand images 
+with 5 human generated captions per image. 
+
+Our model will consist of a VGG16 pretrained image encoder, 
+a vector encoder, and a lstm text decoder.
+
+First, lets import all the modules we will need
+"""
+import ludwig
+import tensorflow as tf
+import os
+import pandas as pd
+import numpy as np
+from ludwig.api import LudwigModel
+import logging
+from tqdm import tqdm
+
+"""
+Let's load the dataset which can be done in two lines
+using Ludwig's datasets module. This gives us three dataframes
+for our training, validation, and test sets
+"""
+from ludwig.datasets import flickr8k
+
+train, vali, test = flickr8k.load(split=True)
+
+"""
+We will now load the pre-trained VGG16 model from keras 
+to help us process our images. Since our dataset is relatively small, 
+it would be relatively ineffective to train our own image encoder from scratch
+"""
+from keras.applications import VGG16
+from keras import models
+
+modelvgg = VGG16(include_top=True,weights='imagenet')
+modelvgg = models.Model(inputs=modelvgg.inputs, outputs=modelvgg.layers[-2].output)
+
+"""
+Here we will define a function that will use the pretrained model to extract 
+features from the images in our dataset. Every image will be converted into 
+a size-4096 vector encoding.
+"""
+from keras.preprocessing.image import load_img, img_to_array
+from keras.applications.vgg16 import preprocess_input
+
+def extract_features(dataset, model):
```

Comment:
This is cool, I like how this example shows how Ludwig can be extended / combined with other tools.  This makes me think about how we would accomplish a similar result through adding custom encoder / preprocessing.  Today I think it would be more effort to do this by extending Ludwig classes, but ideally we'd like to get to a point where this could be done as part of the LudwigModel itself (which would allow us to do things like distribute using Dask).  @w4nderlust what do you think?

Reply:
Yea I was thinking about that too, and there seems to have been an older conversation on adding pretrained image encoders to ludwig at #746 

I can try to help with that; I'm currently looking at ludwigs docs and previous examples to see how that might be done

Also fyi, there is a list of available pretrained image models through keras at https://keras.io/api/applications/.

Reply:
@tgaddair I chatted with @michaelbzhu about it, i suggested to add an encoder similar to what we did for the HF text encoders, which also contains specific preprocessing instructions. That could be a nice way to add additional pretrained image encoders.